src/axelar/AxelarTransceiver.sol

```diff
@@ -0,0 +1,182 @@
+
+    constructor(
+        address _gateway,
+        address _gasService,
+        address _manager
+    ) AxelarGMPExecutable(_gateway) Transceiver(_manager) {
+        gasService = IAxelarGasService(_gasService);
+    }
+
+    /**
+     * Set the bridge manager contract address
+     * @param chainId The chainId of the chain. This is used to identify the chain in the EndpointManager.
+     * @param chainName The chainName of the chain. This is used to identify the chain in the AxelarGateway.
+     * @param transceiverAddress The address of the tranceiver on the other chain, in the axelar accepted format.
+     */
+    function setAxelarChainId(
+        uint16 chainId,
+        string calldata chainName,
+        string calldata transceiverAddress
+    ) external onlyOwner {
+        AxelarTransceiverStorage storage slot = _storage();
+        slot.idToAxelarChainId[chainId] = chainName;
+        slot.axelarChainIdToId[chainName] = chainId;
+        slot.idToTransceiverAddress[chainId] = transceiverAddress;
+        slot.transceiverAddressToId[transceiverAddress] = chainId;
+    }
+
+    /// @notice Fetch the delivery price for a given recipient chain transfer.
+    /// @param recipientChainId The Wormhole chain ID of the target chain.
+    /// param instruction An additional Instruction provided by the Transceiver to be
+    ///        executed on the recipient chain.
+    /// @return deliveryPrice The cost of delivering a message to the recipient chain,
+    ///         in this chain's native token.
+    function _quoteDeliveryPrice(
+        uint16 recipientChainId,
+        TransceiverStructs.TransceiverInstruction memory /*instruction*/
+    ) internal view virtual override returns (uint256) {
+        // Use the gas estimation from gas service
+        AxelarTransceiverStorage storage slot = _storage();
+        return gasService.estimateGasFee(
+            slot.idToAxelarChainId[recipientChainId],
+            slot.idToTransceiverAddress[recipientChainId],
+            bytes(""),
+            DESTINATION_EXECUTION_GAS_LIMIT,
+            bytes("")
+        );
+    }
+
+    /// @dev Send a message to another chain.
+    /// @param recipientChainId The Wormhole chain ID of the recipient.
+    /// @param deliveryPayment the amount of native tokens to be used as gas for delivery.
+    /// @param recipientNttManagerAddress the address of the NttManager to receive this message.
+    /// @param refundAddress the address to receive the gas refund if overpayed.
+    /// @param nttManagerMessage A message to be sent to the nttManager on the recipient chain.
+    function _sendMessage(
+        uint16 recipientChainId,
+        uint256 deliveryPayment,
+        address, /*caller*/
+        bytes32 recipientNttManagerAddress,
+        bytes32 refundAddress,
+        TransceiverStructs.TransceiverInstruction memory, /*transceiverInstruction*/
+        bytes memory nttManagerMessage
+    ) internal virtual override onlyNttManager {
```

Comment:
I don't think this needs the `onlyNttManager` modifier since the exposed `sendMessage` in `Transceiver` already has this

Reply:
goot catch